microsoft-edge/progressive-web-apps-chromium/how-to/service-workers.md

```diff
@@ -1,34 +1,66 @@
+ms.date: 11/28/2022
 ---
-# Use Service Workers to manage network requests and push notifications
+# Use Service Workers to manage network requests
 
-Service Workers are a special type of Web Worker with the ability to intercept, modify, and respond to all network requests using the `Fetch` API.  Service Workers can access the `Cache` API, and asynchronous client-side data stores, such as `IndexedDB`, to store resources.
+Service Workers are a special type of [Web Workers](https://developer.mozilla.org/docs/Web/API/Web_Workers_API) with the ability to intercept, modify, and respond to network requests using the `Fetch` API.  Service Workers can access the `Cache` API, and asynchronous client-side data stores, such as `IndexedDB`, to store resources.
+
+Service Workers can make your PWA faster by caching resources locally and they can also make your PWA more reliable by making it network-independent.
+
+The first time a user accesses your PWA, its Service Worker is installed. The Service Worker then runs in parallel to your app, and can continue doing work even when your app is not running.
+
+Service Workers are mostly responsible for intercepting, modifying, and responding to network requests They can be alerted when the app tries to load a resource from the server, or sends a request to get data from the server. When this happens, a Service Worker can decide to let the request go to the server, or intercept it and return a response from the cache instead.
+
+![High level architecture diagram showing that the service worker is in between the app, on one side, and the network and cache store, on the other side.](../media/sw-architecture.png)
 
 
 <!-- ====================================================================== -->
-## Registering a Service Worker
+## Register a Service Worker
 
 Similar to other Web Workers, Service Workers must exist in a separate file. You reference this file when registering the Service Worker, as shown in the following code:
 
 ```javascript
-if ( "serviceWorker" in navigator ) {
-    navigator.serviceWorker.register( "/serviceworker.min.js" );
+if ("serviceWorker" in navigator) {
+    navigator.serviceWorker.register( "/serviceworker.js");
 }
 ```
 
-Modern browsers provide different levels of support for Service Workers. As such, it's a good practice to test for the existence of the `serviceWorker` object before running any Service Worker-related code. In the above code, a Service Worker is registered using the `serviceworker.min.js` file located at the root of the site.
+The web browser that's running your PWA may provide different levels of support for Service Workers. Additionally, the context in which your PWA is running may not be secure. As such, it's a good practice to test for the existence of the `navigator.serviceWorker` object before running any Service Worker-related code. In the above code, a Service Worker is registered using the `serviceworker.js` file located at the root of the site.
```

Comment:
```suggestion
The web browser that's running your PWA may provide different levels of support for Service Workers. Also, the context in which your PWA is running might not be secure. As such, it's a good practice to test for the existence of the `navigator.serviceWorker` object before running any Service Worker-related code. In the above code, a Service Worker is registered by using the `serviceworker.js` file that's located at the root of the site.
```
Acrolinx would suggest 'also'

'may' is subtle, no hard & fast rule.  I probably never use it but write in a more forceful less gentle way.
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/can-may - sometimes this applies: "Don't use may, which might be interpreted as providing permission."